boards/riscv/tlsr9518adk80d/pinmux.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gpio.h>
+#include <drivers/pinmux.h>
+
+
+/**
+ *  @brief  Define UART0 pins : TX(PA3 PB2 PD2), RX(PA4 PB3 PD3)
+ */
+#define UART0_TX_PA3            GPIO_PA3
```

Comment:
Can we encode these in devicetree.  Take a look at what we do on STM32 or NXP Kinetsis devicetrees for an example.

Reply:
During the development, I saw that some vendors are using "devicetree" and some vendors are using "pinmux.c" solution. 
I decided to go with "pinmux.c" solution because I preflare to have all pins configuration in one place. 
As I see currently 62 boards are using this solution.

Is it something what must be fixed? Or, it is just a recommendation?

Reply:
This should be addressed as the direction we are moving towards is utilizing devicetree for this purpose.